primitives/xcm/src/barriers.rs

```diff
@@ -62,8 +61,9 @@ impl<T: Contains<MultiLocation>> ShouldExecute for AllowTopLevelPaidExecutionDes
 			BuyExecution {
 				weight_limit: Limited(ref mut weight),
 				..
-			} if *weight >= max_weight => {
-				*weight = max_weight;
+			} if weight.all_gte(max_weight) => {
```

Comment:
should this be `any_gte` or should we handle the different weight components separately here? Otherwise, I think you could circumvent one weight component (e.g. proof size) by having a low-enough other component (ref time).

Reply:
After talking with @girazoki this is how it should be implemented. The condition is part of the `match` part, meaning if one of the components overflow but the other doesn't it will throw an `Err`.